firefox_puppeteer/base.py

```diff
@@ -28,15 +28,31 @@ def get_marionette(self):
 class UIBaseLib(BaseLib):
     """A base class for all UI element wrapper classes inside a chrome window."""
 
-    def __init__(self, marionette_getter, window):
-        BaseLib.__init__(self, marionette_getter)
-
+    def __init__(self, marionette_getter, parent, element):
         # importing globally doesn't work
         from .ui.windows import BaseWindow
-        if not isinstance(window, BaseWindow):
+
+        BaseLib.__init__(self, marionette_getter)
+
+        if isinstance(parent, UIBaseLib):
+            self._parent = parent
+            self._window = parent.window
+        elif isinstance(parent, BaseWindow):
+            self._parent = parent
+            self._window = parent
+        else:
             raise errors.UnexpectedWindowTypeError('Not a valid BaseWindow: "%s"' %
-                                                   window)
-        self._window = window
+                                                   parent)
+
+        self._element = element
+
+    @property
+    def element(self):
+        return self._element
+
+    @property
+    def parent(self):
+        return self._parent
```

Comment:
If this is all fine we will have a follow-up bug to convert the remaining UI classes to UIBaseLib.